[One Workflow] Fix search in manual run modal

VladimirFilonov opened the pull request:
closes: https://github.com/elastic/security-team/issues/14918

Improvements:
* Show message column in Alert tab
* Transform Index data into DataView to get suggested field
* Map `rule.*` to `kibana.alert.rule.*` for legacy compatibility

VladimirFilonov committed:
Fix search in manual run modal

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/367812)
* Commit: ee100526b5b2ca9072465889ad7c074218c55fec

### Failed CI Steps
* [Build API Docs](https://buildkite.com/elastic/kibana-pull-request/builds/367812#019adfdf-5fd0-4235-9448-189a7523bf2c)
* [Pick Test Group Run Order](https://buildkite.com/elastic/kibana-pull-request/builds/367812#019adfdf-5fd1-4444-9417-096369c254a9)
* [Pick Test Group Run Order](https://buildkite.com/elastic/kibana-pull-request/builds/367812#019adffc-606f-4bf6-b9ac-bd201cfc4e1a)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/367812#019adfdf-5fd2-4e99-b16b-0b4f634bde5a)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/367812#019adffc-08c8-4059-add0-27f02e4b1ab6)

### History


<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367812","number":367812,"commit":"ee100526b5b2ca9072465889ad7c074218c55fec"}],"number":367812}
buildkite-pr-comment-->

VladimirFilonov committed:
Merge branch 'main' into fix/14918-bug-search-is-not-working-in-manual-run-popup

VladimirFilonov committed:
Merge branch 'main' into fix/14918-bug-search-is-not-working-in-manual-run-popup

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368072)
* Commit: 3592585486dfd92f0ac4091ca46ac4255dd47227

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b5c6f7bd63edaa21e2c75231e04d814cfd23ae60) | [after](https://github.com/elastic/kibana/commit/3592585486dfd92f0ac4091ca46ac4255dd47227) | diff |
| --- | --- | --- | --- |
| `workflowsManagement` | 2.1MB | 2.1MB | +1.7KB |



### History
* :broken_heart: [Build #367857](https://buildkite.com/elastic/kibana-pull-request/builds/367857) failed 01c7ad23b372518750fe80664619d9b40a415c15
* :broken_heart: [Build #367812](https://buildkite.com/elastic/kibana-pull-request/builds/367812) failed ee100526b5b2ca9072465889ad7c074218c55fec

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368072","number":368072,"commit":"3592585486dfd92f0ac4091ca46ac4255dd47227"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367857","number":367857,"commit":"01c7ad23b372518750fe80664619d9b40a415c15"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367812","number":367812,"commit":"ee100526b5b2ca9072465889ad7c074218c55fec"}],"number":368072}
buildkite-pr-comment-->

VladimirFilonov committed:
refactor(workflow_execute_event_form): update alert fetching logic and integrate AlertsSearchBar

- Replaced the existing SearchBar with AlertsSearchBar for improved alert querying.
- Refactored alert fetching to utilize SearchSource, aligning with Discovery's behavior.
- Simplified state management by removing unnecessary data view handling and query transformation.
- Added support for filters and sorting in the alert fetching process.
- Updated time range handling to ensure proper date formatting.

VladimirFilonov committed:
Refactor manual run modal search to match with Discovery's behaviour where possible

VladimirFilonov committed:
Merge remote-tracking branch 'upstream/main' into fix/14918-bug-search-is-not-working-in-manual-run-popup

kibanamachine committed:
Changes from node scripts/lint_ts_projects --fix

kibanamachine committed:
Changes from node scripts/regenerate_moon_projects.js --update

skynetigor reviewed (approved):
LGTM
